modules/taboolaBidAdapter.js

```diff
@@ -60,13 +60,13 @@ export const userData = {
 
 export const internal = {
   getPageUrl: (refererInfo = {}) => {
-    return refererInfo.page || getWindowSelf().location.href;
+    return refererInfo?.page || getWindowSelf().location?.href;
   },
   getReferrer: (refererInfo = {}) => {
-    if (refererInfo.ref) {
+    if (refererInfo?.ref) {
```

Comment:
@ahmadlob here we don't need the ?, because we check if (refererInfo.ref) 
we can change the if else to be same as the page url

Reply:
the If checks if "refererInfo.ref" is null. the chaining I added -> If "refererInfo" is null - evaluate "undefined" and the general condition to be falsy. 
It prevent an error to be thrown when refererIndo is undefined.

Reply:
changed it to be much more shorter